server/src/test/java/org/elasticsearch/search/dfs/DfsPhaseTests.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.dfs;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.KnnFloatVectorField;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.KnnFloatVectorQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.tests.index.RandomIndexWriter;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.search.internal.ContextIndexSearcher;
+import org.elasticsearch.search.profile.Profilers;
+import org.elasticsearch.search.profile.SearchProfileDfsPhaseResult;
+import org.elasticsearch.search.profile.query.CollectorResult;
+import org.elasticsearch.search.profile.query.QueryProfileShardResult;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.threadpool.TestThreadPool;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public class DfsPhaseTests extends ESTestCase {
+
+    ThreadPoolExecutor threadPoolExecutor;
+    private TestThreadPool threadPool;
+
+    @Before
+    public final void init() {
+        int numThreads = randomIntBetween(2, 4);
+        threadPool = new TestThreadPool(DfsPhaseTests.class.getName());
+        threadPoolExecutor = EsExecutors.newFixed(
+            "test",
+            numThreads,
+            10,
+            EsExecutors.daemonThreadFactory("test"),
+            threadPool.getThreadContext(),
+            randomBoolean()
+        );
+    }
+
+    @After
+    public void cleanup() {
+        threadPoolExecutor.shutdown();
+        terminate(threadPool);
+    }
+
+    public void testSingleKnnSearch() throws IOException {
```

Comment:
Thanks for adding this test! Do we need to test this specific method or could we instead test `executeKnnVectorQuery` directly? Maybe as a follow-up we can unit test also the terms statistics collection? we should have totally written these tests 10 years ago but it seems like we haven't :)